rfd/0167-console-service.md

```diff
@@ -0,0 +1,225 @@
+
+// While making requests using the client, the users will still require to
+// provide the full URL since the client always validate them.
+res, err := client.Get("http://console-service/debug/pprof/heap")
+```
+</details>
+
+#### Changing log level command
+
+Changes the application log level.
+
+`teleport [-c config-path] console set-log-level [LEVEL]`
+
+|Flag|Description|Default value|
+|----|-----------|-------------|
+|`-c,--config`|Teleport configuration path.|`/etc/teleport.yaml`|
+|`LEVEL`|Log level (case-insensitive). Any of: `TRACE`, `DEBUG`, `INFO`, `WARN`, `ERROR`|Required|
+
+Usage examples:
+```bash
+$ teleport console set-log-level DEBUG
+$ teleport -c /random/teleport.yaml console set-log-level INFO
+```
+
+#### Capture `pprof` profiles command
+
+Export the application profile (`pprof` format).
+
+`teleport [-c config-path] console profile [--seconds=] [PROFILE_NAME]`
+
+The `PROFILE_NAME` values follow the Golang's definition on `runtime.Profile`
+plus the profiles defined by `net/http/pprof`:
+- `allocs`: A sampling of all past memory allocations
+- `block`: Stack traces that led to blocking on synchronization primitives
+- `cmdline`: The command line invocation of the current program
+- `goroutine`: Stack traces of all current goroutines. Use debug=2 as a query parameter to export in the same format as an unrecovered panic.
+- `heap`: A sampling of memory allocations of live objects. You can specify the gc GET parameter to run GC before taking the heap sample.
+- `mutex`: Stack traces of holders of contended mutexes
+- `profile`: CPU profile. You can specify the duration in the seconds GET parameter. After you get the profile file, use the go tool pprof command to investigate the profile.
+- `threadcreate`: Stack traces that led to the creation of new OS threads
+- `trace`: A trace of execution of the current program. You can specify the duration in the seconds GET parameter. After you get the trace file, use the go tool trace command to investigate the trace.
+
+In addition to those profiles, the command will also support a `default` profile
+which will include `heap`, `profile` and `goroutine` profiles.
+
+Note: `--seconds` argument has the same effect as
+[`seconds` query string](https://pkg.go.dev/net/http/pprof#hdr-Parameters).
+This parameter max value will be set to the HTTP server write timeout value.
+
+|Flag|Description|Default value|
+|----|-----------|-------------|
+|`-c,--config`|Teleport configuration path.|`/etc/teleport.yaml`|
+|`-s,--seconds`|For CPU and trace profiles, profile for the given duration. For other profiles, return a delta profile.|None|
+|`PROFILE_NAME`|Profile to be exported. Any of: `default`, `allocs`, `block`, `cmdline`, `goroutine`, `heap`, `mutex`, `profile`, `threadcreate`, `trace`|`default`|
+
+### Security
+
+Items listed on this section are have their impact limited due to the fact that
+the service will not be exposed outsite the machine/container running the
+Teleport instance.
+
```

Comment:
It's worth also mentioning that there is potential information disclosure through this tool.  For example the ability to profile could be combined with an attack to understand the state of accounts or other cluster state.